backend/src/main/java/de/neuefische/backend/service/UtilService.java

```diff
@@ -1,25 +1,62 @@
 package de.neuefische.backend.service;
 
+import de.neuefische.backend.dto.AppointmentDto;
 import de.neuefische.backend.dto.DoctorDto;
+import de.neuefische.backend.model.Appointment;
 import de.neuefische.backend.model.Doctor;
 import org.springframework.stereotype.Service;
 
+import java.util.List;
+
 @Service
 public class UtilService {
 
+    public Appointment mapAppointmentDtoToAppointment(AppointmentDto appointmentDto) {
+        return Appointment.builder()
+                .date(appointmentDto.getDate())
+                .reasonForVisit(appointmentDto.getReasonForVisit())
+                .notes(appointmentDto.getNotes())
+                .build();
+    }
+
+
     public Doctor mapDoctorDtoToDoctor(DoctorDto doctorDto) {
+
+        AppointmentDto appointmentDto = doctorDto.getAppointmentDto();
+
+        if (appointmentDto != null) {
+            Appointment appointment = mapAppointmentDtoToAppointment(appointmentDto);
+
             return Doctor.builder()
```

Comment:
too much redundance below here. Better: 
```suggestion
    Doctor doctor = Doctor.builder().firstName....emailAddress(doctorDto.getEmailAddress()).build()
    if(appointmentDto != null){mapping...
    doctor.setAppointments(appointment}
    return doctor
```